src/gen/conditions.rs

```diff
@@ -276,23 +276,31 @@ pub fn parse_args(
         }
         ASSERT_BEFORE_SECONDS_RELATIVE => {
             maybe_check_args_terminator(a, c, flags)?;
-            let seconds = parse_seconds(a, first(a, c)?, ErrorCode::AssertBeforeSecondsRelative)?;
-            Ok(Condition::AssertBeforeSecondsRelative(seconds))
+            match parse_positive_seconds(a, first(a, c)?, ErrorCode::AssertBeforeSecondsRelative)? {
+                None => Ok(Condition::Skip),
+                Some(seconds) => Ok(Condition::AssertBeforeSecondsRelative(seconds)),
+            }
         }
         ASSERT_BEFORE_SECONDS_ABSOLUTE => {
             maybe_check_args_terminator(a, c, flags)?;
-            let seconds = parse_seconds(a, first(a, c)?, ErrorCode::AssertBeforeSecondsAbsolute)?;
-            Ok(Condition::AssertBeforeSecondsAbsolute(seconds))
+            match parse_positive_seconds(a, first(a, c)?, ErrorCode::AssertBeforeSecondsAbsolute)? {
+                None => Ok(Condition::Skip),
+                Some(seconds) => Ok(Condition::AssertBeforeSecondsAbsolute(seconds)),
+            }
         }
         ASSERT_BEFORE_HEIGHT_RELATIVE => {
             maybe_check_args_terminator(a, c, flags)?;
-            let height = parse_height(a, first(a, c)?, ErrorCode::AssertBeforeHeightRelative)?;
-            Ok(Condition::AssertBeforeHeightRelative(height))
+            match parse_positive_height(a, first(a, c)?, ErrorCode::AssertBeforeHeightRelative)? {
+                None => Ok(Condition::Skip),
+                Some(height) => Ok(Condition::AssertBeforeHeightRelative(height)),
+            }
         }
         ASSERT_BEFORE_HEIGHT_ABSOLUTE => {
             maybe_check_args_terminator(a, c, flags)?;
-            let height = parse_height(a, first(a, c)?, ErrorCode::AssertBeforeHeightAbsolute)?;
-            Ok(Condition::AssertBeforeHeightAbsolute(height))
+            match parse_positive_height(a, first(a, c)?, ErrorCode::AssertBeforeHeightAbsolute)? {
+                None => Ok(Condition::Skip),
+                Some(height) => Ok(Condition::AssertBeforeHeightAbsolute(height)),
+            }
```

Comment:
these 4 hunks are the main changes to production code, altering the behavior of the new conditions